domain/ide/workspace/ui/diagnostics_overview.go

```diff
@@ -0,0 +1,204 @@
+
+func init() {
+	logger := config.CurrentConfig().Logger().With().Str("method", "ui.init").Logger()
+	funcMap := map[string]any{}
+	var err error
+	diagnosticsOverviewTemplate, err = template.New("diagnosticsOverviewTemplate").Funcs(funcMap).Parse(diagnosticsOverviewTemplatePath)
+	if err != nil {
+		logger.Err(err).Msg("failed to parse template")
+		return
+	}
+}
+
+type TemplateData struct {
+	// Root nodes
+	RootNodes []Node
+	// File node with underlying slice of issue nodes
+	Issues               map[Node][]Node
+	Styles               template.CSS
+	Nonce                string
+	DeltaFindingsEnabled bool
+}
+
+// Node represents a tree node
+type Node struct {
+	Icon           template.HTML
+	Text           template.HTML
+	ProductEnabled bool
+}
+
+func SendDiagnosticsOverview(c *config.Config, p product.Product, issuesByFile snyk.IssuesByFile, notifier notification.Notifier) {
+	logger := c.Logger().With().Str("method", "ui.SendDiagnosticsOverview").Logger()
+	if p != "" {
+		rootNodes := getRootNodes(c, p, issuesByFile)
+		nonce, err := generateSecurityNonce()
+		if err != nil {
+			logger.Err(err).Msgf("Failed to generate nonce")
+			return
+		}
+
+		fileNodes := getFileNodes(issuesByFile)
+
+		data := TemplateData{
+			RootNodes:            rootNodes,
+			Issues:               fileNodes,
+			Styles:               template.CSS(diagnosticsOverviewTemplateCSS),
+			Nonce:                nonce,
+			DeltaFindingsEnabled: c.IsDeltaFindingsEnabled(),
+		}
+
+		var htmlBuffer bytes.Buffer
+		if err = diagnosticsOverviewTemplate.Execute(&htmlBuffer, data); err != nil {
+			logger.Error().Msgf("Failed to generate tree htmlBuffer with tree template: %v", err)
+			return
+		}
+
+		diagnosticsOverviewParams := types.DiagnosticsOverviewParams{Product: p.ToProductCodename(), Html: htmlBuffer.String()}
+		notifier.Send(diagnosticsOverviewParams)
+		logger.Debug().Msgf("sent diagnostics overview htmlBuffer for product %s", p)
+		logger.Trace().
+			Int("issueCount", len(issuesByFile)).
+			Any("diagnosticsOverviewParams", diagnosticsOverviewParams).
+			Msg("detailed tree data")
+	}
+}
+
+func getFileNodes(issuesByFile snyk.IssuesByFile) map[Node][]Node {
+	fileNodes := make(map[Node][]Node)
+	for path, issues := range issuesByFile {
+		fileNode := Node{
+			Icon: getFileTypeIcon(),
+			Text: template.HTML(path),
+		}
+		issueNodes := []Node{}
+		for _, issue := range issues {
+			issueNodes = append(issueNodes, Node{
+				Icon: html.GetSeverityIconSvg(issue),
+				Text: template.HTML(issue.AdditionalData.GetTitle()),
+			})
+		}
+		fileNodes[fileNode] = issueNodes
+	}
+	return fileNodes
+}
+
+func getFileTypeIcon() template.HTML {
+	return ""
+}
+
+// generateSecurityNonce generates a cryptographically secure random nonce.
+// A nonce is used in the Content Security Policy (CSP) to allow specific
+// inline styles and scripts, helping to prevent Cross-Site Scripting (XSS) attacks.
+func generateSecurityNonce() (string, error) {
```

Comment:
If I remember we were also calculating nonces for the other HTML views correct?
If this was done in LS maybe it makes sense to move this somewhere else

Reply:
Yeah, this is a real lazy repeat. I'll DRY it up.

Reply:
Actually, I didn't find repetition, but it's gonna be good to have it in the `html` package anyway. I think, the IaC PR can re-use it.